workers/data_refinery_workers/processors/management/commands/create_compendia.py

```diff
@@ -13,57 +13,70 @@
     Dataset,
     ProcessorJobDatasetAssociation,
     ExperimentOrganismAssociation,
-    OrganismIndex,
     ExperimentSampleAssociation
 )
 from data_refinery_workers.processors import create_compendia
 
 logger = get_and_configure_logger(__name__)
 
 
+def create_job_for_organism(organism=Organism):
+    """Returns a compendia job for the provided organism.
+
+    Fetch all of the experiments and compile large but normally formated Dataset.
+    """
+    data = {}
+    experiments = Experiment.objects.filter(id__in=(ExperimentOrganismAssociation.objects.filter(organism=organism)).values('experiment'))
+    for experiment in experiments:
+        data[experiment.accession_code] = list(experiment.samples.filter(organism=organism).values_list('accession_code', flat=True))
+
+    job = ProcessorJob()
+    job.pipeline_applied = "COMPENDIA"
+    job.save()
+
+    dset = Dataset()
+    dset.data = data
+    dset.scale_by = 'NONE'
+    dset.aggregate_by = 'SPECIES'
+    dset.quantile_normalize = False
+    dset.save()
+
+    pjda = ProcessorJobDatasetAssociation()
+    pjda.processor_job = job
+    pjda.dataset = dset
+    pjda.save()
+
+    return job
+
+
 class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument(
-            "--organism",
+            "--organisms",
             type=str,
-            help=("Name of organism"))
+            help=("Comma separated list of organism names."))
 
     def handle(self, *args, **options):
-        """ For every (or a supplied) organism, fetch all of the experiments and compile large but normally formated Dataset.
+        """Create a compendium for one or more organisms.
 
-        Send all of them to the Smasher. Smash them. Retrieve manually as desired.
+        If --organism is supplied will immediately create a compedium
+        for it. If not a new job will be dispatched for each organism
+        with enough microarray samples.
         """
-
-        dataset_ids = []
-
-        if options["organism"] is None:
+        if options["organisms"] is None:
             all_organisms = Organism.objects.all()
```

Comment:
Where do we check that the organisms have enough microarray samples?

Reply:
In the compendia job.

Reply:
Is it worth checking here to avoid dispatching these jobs?

Reply:
Mmmm I think it might be more trouble than it's worth.